lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchDriverConfigurationProviderImpl.java

```diff
@@ -82,9 +99,18 @@ public Configuration getDriverConfiguration(final URI jobFolder,
                                               final String clientRemoteId,
                                               final String jobId,
                                               final Configuration applicationConfiguration) {
+
+
+    String[] ports = {"2000", "2001" };
+
+    final String availablePortsList = StringUtils.join(ports, ",");
     return Configurations.merge(
         AzureBatchDriverConfiguration.CONF.getBuilder()
-            .bindImplementation(CommandBuilder.class, this.commandBuilder.getClass()).build()
+            .bindImplementation(CommandBuilder.class, this.commandBuilder.getClass())
+            .bindImplementation(LocalAddressProvider.class, ContainerBasedLocalAddressProvider.class)
+            .bindNamedParameter(TcpPortList.class, availablePortsList)
+            .bindImplementation(TcpPortProvider.class, ListTcpPortProvider.class)
+            .build()
```

Comment:
Step 1, I would say remove these hard coded ports so you know where you stand. Same goes for the RuntimeConfiguration.